pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1930,7 +1965,18 @@ func handleModify(ctx *domainContext, key string,
 }
 
 func updateStatusFromConfig(status *types.DomainStatus, config types.DomainConfig) {
-	status.VirtualizationMode = config.VirtualizationMode
+	virtualizationMode := config.VirtualizationMode
+	switch config.VirtualizationMode {
+	case types.PV:
+		// Do nothing
+	case types.HVM:
+		// Do nothing
+	case types.FML:
+		// Do nothing
+	default:
+		virtualizationMode = types.PV
+	}
```

Comment:
Hmm.. This logic is spread multiple times.. its better to add a method to config object?
   status.virtualizationMode = config.VirtualizationModeOrDefault()

Reply:
Good suggestion. I'll do that.